pkg/apply/reconcilers.go

```diff
@@ -116,7 +116,7 @@ func reconcileService(oldObj, newObj runtime.Object) (bool, error) {
 	return false, nil
 }
 
-func reconcileJob(oldObj, newObj runtime.Object) (bool, error) {
+func reconcileJob(oldObj, newObj, _ runtime.Object) (bool, error) {
```

Comment:
since youre only using the new prune obj here why not just round trip in this func and then we won't need to break the `Reconciler` api?